doc/staterecovery_sequencediagram.txt

```diff
@@ -0,0 +1,32 @@
+title State Recovery Procedure
+
+participant syncerd
+// TODO: remove walletd once it's stateless
+participant walletd
+participant swapd
+participant checkpointd
+participant farcasterd
+participant cli
+participant peerd
+
+== State Recovery Procedure
+entryspacing 0.8
+farcasterd -> checkpointd : launch
+checkpointd -> farcasterd : RecoverableStates
+farcasterd -> cli : RecoverableStates
+cli -> farcasterd : RecoverSelectedState
+farcasterd -> walletd : launch
+farcasterd -> swapd : launch
+farcasterd -> syncerd : launch
+farcasterd -> peerd : launch
```

Comment:
here all the components are relaunched in a coupled manner, which probably doesnt reflect single component recovery, thus i guess if i ignore the lines related to other components and recover a single one, everything works, right?